chapter05/spg/exercise3.pl

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+# There are no comments!
+my $data = {};
+my $output = "coconet_sorted.dat";
+
+while ( <> ) {
+  my ( $source, $dest, $transfer ) = split;
+  $data->{$source}->{$dest} += $transfer;
+}
+
+open( my $file, '>', $output ) or die "could not create file '$output' $!";
+
+foreach my $source ( sort { $a cmp $b } keys $data ) {
```

Comment:
You didn't strictly need the sort block here, since that's the default behaviour.